packages/ozone/src/tag-service/embed-content-type-tagger.ts

```diff
@@ -0,0 +1,54 @@
+import { AppBskyEmbedImages, AppBskyEmbedRecordWithMedia } from '@atproto/api'
+import { langLogger as log } from '../logger'
+import { ContentTagger } from './content-tagger'
+import { ids } from '../lexicon/lexicons'
+
+export class EmbedContentTypeTagger extends ContentTagger {
+  tagPrefix = 'embed-content-type:'
+
+  isApplicable(): boolean {
+    return (
+      !!this.subjectStatus &&
+      !this.tagAlreadyExists() &&
+      this.subject.isRecord() &&
+      this.subject.parsedUri.collection === ids.AppBskyFeedPost
+    )
+  }
+
+  async getTags(): Promise<string[]> {
+    try {
+      if (!this.isApplicable()) {
+        return []
+      }
+      const recordValue = await this.getRecordValue()
+      if (!recordValue) {
+        return []
+      }
+      const tags: string[] = []
+      if (
+        AppBskyEmbedImages.isView(recordValue) &&
```

Comment:
I think we want this:
```suggestion
        AppBskyFeedPost.isRecord(recordValue) &&
```

This functionality should be fairly testable, I think—maybe should pass a post record through it.

Reply:
yeah good idea, was easier to get the right checks through a test. LMK if the current checks look right. my reasoning was, we wanna check if the record is a post with media embed first. and then, apply tags based on the type of media.